enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,546 @@
+* overlappingrangeipreservations.whereabouts.cni.cncf.io
+* volumesnapshotclasses.snapshot.storage.k8s.io
+* volumesnapshotcontents.snapshot.storage.k8s.io
+* volumesnapshots.snapshot.storage.k8s.io
+
+This limitation is unlikely to be triggered by manifests created by
+the OpenShift installer, but we cannot control what extra manifests
+users add to their deployment. Users need to be made aware of this
+limitation and encouraged to avoid creating custom manifests using
+CRDs installed by cluster operators instead of the
+`cluster-version-operator`.
+
+#### Post-reboot service makes single-node deployments different from multi-node clusters
+
+The `post-reboot` service is only used with single-node
+deployments. This makes those deployments look different in a way that
+may lead to confusion when debugging issues on the cluster. To
+mitigate this, we can add documentation to the troubleshooting guide
+to explain the service and its role in the cluster.
+
+#### Bootstrap logs retention
+ 
+ Due to the bootstrap-in-place behavior all bootstrap artifacts
+ will be lost once the bootstrap the node reboots.
+In a regular installation flow the bootstrap node goes down only once
+ the control plane is running, and the bootstrap node served its purpose.
+In case of bootstrap in place things can go wrong after the reboot.
+The bootstrap logs can aid in troubleshooting a subsequently failed install. 
+
+Mitigation by gathering the bootstrap logs before reboot.
+ bootstrap will gather logs from itself using /usr/local/bin/installer-gather.sh.
+ Once gathering is complete, the bundle will be added to the master ignition,
+ thus making the bootstrap logs available from the master after reboot.
+ The log bundle will be deleted once the installation completes.
+ 
+The installer `gather` command works as usual before the reboot.
+We will add a new script called installer-master-bootstrap-in-place-gather.sh.
+ This script will be delivered to the master using via Ignition to the same
+ location where the bootstrap node usually has installer-gather.sh.
+The installer-master-bootstrap-in-place-gather.sh, will be called by the
+ `openshift-install gather` command that believes its collecting logs from a bootstrap node.
+The script however behaves slightly different, instead of collecting bootstrap logs
+and then remotely running /usr/local/bin/installer-masters-gather.sh on all master nodes,
+ the script will collect the bootstrap logs from the bundle copied to via the master-ignition,
+ and collect master logs by running /usr/local/bin/installer-masters-gather.sh directly
+ on itself. The final archiving of all the logs into the home directory,
+ exactly the same as /usr/local/bin/installer-gather.sh.
+
+## Design Details
+
+### Open Questions
+
+1. How will the user specify custom configurations, such as static IPs?
+2. Number of revisions for the control plane - do we want to make changes to the bootstrap static pods to make them closer to the final ones?
```

Comment:
Is this optimization possible in the multi-node case as well? Seem orthogonal to this enhancement, no?

Reply:
Yes the service should delete itself

Reply:
Right, this optimization should work the same for multi-node.